lms/djangoapps/ytp_api/serializers.py

```diff
@@ -0,0 +1,104 @@
+# coding=utf-8
+"""
+Serializer for validate data.
+"""
+
+from django.contrib.auth.models import User
+from django.core.exceptions import ValidationError
+from rest_framework.serializers import HyperlinkedModelSerializer
+
+from openedx.core.djangoapps.user_api.accounts.api import check_account_exists
+from openedx.core.djangoapps.user_api.serializers import ReadOnlyFieldsSerializerMixin
+from student.models import CourseEnrollment, LanguageProficiency, UserProfile
+
+
+class UserSerializer(HyperlinkedModelSerializer):
+    """
+    Class that serializes the information of User model.
+
+    Update and validate User information.
+    """
+
+    class Meta(object):
+        model = User
+        read_only_fields = ('password', 'username')
```

Comment:
Because UserSerializer always works with username as a required parameter if We don't add a username to the read_only_fields. And If We use UserSerializer in Patch method for updating part of users data without the username, UserSerializer makes an error like this: "Username is a required parameter". 
I added a username to the read_only_fields and I used another serializer (UserUsernameSerializer) to validate username.

Reply:
Got it, thanks.